implementation/src/main/java/io/smallrye/openapi/runtime/scanner/dataobject/TypeResolver.java

```diff
@@ -76,37 +203,201 @@ public Type getResolvedType(Type fieldType) {
+    }
+
+    private static Map<ClassInfo, Type> inheritanceChain(AugmentedIndexView index,
+            ClassInfo klazz,
+            Type type) {
+
+        Map<ClassInfo, Type> chain = new LinkedHashMap<>();
+
+        do {
+            chain.put(klazz, type);
+        } while ((type = klazz.superClassType()) != null && (klazz = index.getClass(type)) != null);
+
+        return chain;
+    }
 
-            currentType = currentClass.superClassType();
+    private static void scanField(Map<String, TypeResolver> properties, FieldInfo field, Deque<Map<String, Type>> stack) {
+        String propertyName = field.name();
+        String methodSuffix = Character.toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
 
-            if (currentType == null) {
-                break;
+        // Consider only using fields that are public?
+        if (properties.containsKey(propertyName)) {
+            TypeResolver resolver = properties.get(propertyName);
+
+            if (resolver.getField() == null && (Modifier.isPublic(field.flags()) || Modifier.isProtected(field.flags()))) {
+                /*
+                 * Field is declared in parent class and the getter/setter was
+                 * declared/overridden in child class
+                 */
+                resolver.setField(field);
             }
+        } else {
+            TypeResolver resolver = new TypeResolver(field, new ArrayDeque<>(stack));
+            ClassInfo declaringClass = field.declaringClass();
+            resolver.setReadMethod(declaringClass.method("get" + methodSuffix));
+            resolver.setWriteMethod(declaringClass.method("set" + methodSuffix, field.type()));
+            properties.put(propertyName, resolver);
+        }
+    }
+
+    private static void scanMethod(Map<String, TypeResolver> properties, MethodInfo method, Deque<Map<String, Type>> stack) {
+        String methodName = method.name();
+        Type returnType = method.returnType();
+        Type propertyType;
+
+        if (methodName.startsWith("get") &&
+                returnType.kind() != Type.Kind.VOID &&
+                method.parameters().isEmpty()) {
+            propertyType = returnType;
+        } else if (methodName.startsWith("set") &&
+                returnType.kind() == Type.Kind.VOID &&
+                method.parameters().size() == 1) {
+            propertyType = method.parameters().get(0);
+        } else {
+            propertyType = null;
+        }
+
+        if (propertyType != null) {
+            updateTypeResolvers(properties, stack, method, propertyType);
+        }
+    }
+
+    private static void updateTypeResolvers(Map<String, TypeResolver> properties,
+            Deque<Map<String, Type>> stack,
+            MethodInfo method,
+            Type propertyType) {
+        String methodName = method.name();
+        String propertyName = Character.toLowerCase(methodName.charAt(3)) + methodName.substring(4);
```

Comment:
property names can be customized through `javax.json.bind.annotation.JsonbProperty` or `com.fasterxml.jackson.annotation.JsonProperty` annotations

Reply:
Also `javax.xml.bind.annotation.XmlAttribute` / `javax.xml.bind.annotation.XmlElement` because Jackson also recognizes those annotations. Or just one of them, not sure right now.

Reply:
Added support for ordering by customized names and using customized names via JAXB element and attribute annotations. Note that ordering by Jsonb and JAXB customized names may not be how those frameworks serialize objects.